src/tracks/Pattern.cpp

```diff
@@ -752,8 +752,6 @@ PatternView::~PatternView()
 
 void PatternView::update()
 {
-	m_pat->changeLength( m_pat->length() );
```

Comment:
About the new bug, the pattern length is not being updated. Nevertheless, this `m_pat->changeLength( m_pat->length() );` line is not suitable, since it would be a model update from a view update. The fix should update the model from related model changes.